x/tradeshield/keeper/msg_server_execute_orders.go

```diff
@@ -67,10 +72,17 @@ func (k msgServer) ExecuteOrders(goCtx context.Context, msg *types.MsgExecuteOrd
 		}
 
 		// return the error if any
+		// log the error if any
 		if err != nil {
-			return nil, err
+			// Add log about error or not executed
+			perpLog = append(perpLog, fmt.Sprintf("Perpetual order Id:%d cannot be executed due to err: %s", perpetualOrderId, err.Error()))
 		}
 	}
 
+	ctx.EventManager().EmitEvent(sdk.NewEvent(types.TypeEvtExecuteOrders,
+		sdk.NewAttribute("spot_orders", strings.Join(spotLog, "\n")),
+		sdk.NewAttribute("perpetual_orders", strings.Join(perpLog, "\n")),
+	))
```

Comment:
@amityadav0 can you add unit tests that confirm that given the edge case scenarios we are getting those logs as event attributes fired?